enhancements/network/dual-stack.md

```diff
@@ -0,0 +1,236 @@
+---
+title: Dual-Stack Networking
+authors:
+  - "@danwinship"
+reviewers:
+  - TBD
+  - "@dcbw"
+  - "@russellb"
+approvers:
+  - TBD
+creation-date: 2020-04-23
+last-updated: 2020-04-23
+status: implementable
+---
+
+# Dual-Stack Networking
+
+This covers adding dual-stack networking support via ovn-kubernetes in OCP.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Customers need IPv6 support. Customers still need IPv4 support too
+though. Dual-stack FTW!
+
+At the moment we expect dual-stack to be more popular with customers
+than single-stack IPv6, even for customers who mostly only want IPv6,
+because in many cases you end up needing 99% IPv6 and 1% IPv4, and
+single-stack IPv6 Kubernetes does not allow that. (You end up needing
+hacks outside of Kubernetes's view, like Multus-based interfaces.)
+
+## Motivation
+
+Customers have requested this functionality.
+
+### Goals
+
+- All known upstream Kubernetes dual-stack blockers will be resolved.
+  (eg, the Service `ipFamily` problem that blocked us from enabling
+  the `IPv6DualStack` feature gate in 4.4).
+
+- OVN-Kubernetes will support dual-stack networking according to the
+  model upstream Kubernetes finalizes
+
+- We enable the `IPv6DualStack` feature gate by default
+
+- Other OCP components that need to be updated to be dual-stack aware
+  will be updated. (It is not clear how much actual work is needed
+  here. We are currently researching.)
+
+- We will have CI/cluster-bot support for dual-stack clusters on
+  platforms that we support for customers (ie, bare-metal).
+
+- Assuming that the necessary CloudProvider work is completed
+  upstream, we will have CI/cluster-bot support for dual-stack
+  clusters on at least one cloud platform. (If the bare-metal solution
+  is not sufficiently useful/available, we might do some work to help
+  complete the upstream CloudProvider work.)
+
+### Non-Goals
+
+- Dual-stack support for openshift-sdn or Kuryr
+
+- Officially supporting dual-stack for third-party CNI plugins
+
+## Proposal
+
+### Implementation Details/Notes/Constraints
+
+#### Early CI
+
+To avoid the problems we had with single-stack IPv6, we will get CI
+set up early, to ensure that regressions do not occur in components we
+are not directly paying attention to. Note that we do not expect to do
+any work on a forked branch of OCP as we did for single-stack IPv6.
+
+Being able to bring up a dual-stack cluster at all requires enabling
+the `IPv6DualStack` feature gate in `kubelet`, `kube-apiserver`, and
+`kube-controller-manager`. We tried to do this for all clusters in 4.4
```

Comment:
IIRC kube-scheduler too.

Reply:
cluster-kube-scheduler-operator has the "make sure feature gates are as expected" unit test so it needed to be updated when the default feature gates changed, but it looks like kube-scheduler doesn't pull in any of the code that actually cares about that feature gate.